### PR TITLE
Adds client support for multipart content

### PR DESCRIPTION
**Pull request by dposada**
The PR that consumes this change is https://github.com/twosigma/waiter/pull/75

### CODE REVIEW

**Comment by dposada** [comment]
@pschorf any thoughts on this?

File: src/clj/qbits/jet/client/http.clj [hunk 1]
```diff
@@ -321,6 +323,14 @@
                                           (.add f (name k) (str v)))
                                         f))))
 
+       (when (seq multipart)
+             (.content request
+                       (let [provider (MultiPartContentProvider.)]
+                            (doseq [[k v] multipart]
+                                   (.addFieldPart provider k (StringContentProvider. v) nil))
```
Comment (sradack) [inline src/clj/qbits/jet/client/http.clj]: Why assume that the content `v` is a string?

Comment (dposada) [reply src/clj/qbits/jet/client/http.clj]: Changed it so that the caller provides the `ContentProvider` of their choosing

**Comment by dposada** [comment]
Any more feedback on this?

File: src/clj/qbits/jet/client/http.clj (hunk 1 above)
Comment (sradack) [reply src/clj/qbits/jet/client/http.clj]: There may be some opportunities to improve the API.  For example, if a string is passed as `v`, then maybe you automatically wrap it in a `StringContentProvider`.  Is that what the caller is doing?  Might be nice to have that here.

Comment (dposada) [reply src/clj/qbits/jet/client/http.clj]: Agreed, added the auto-wrapping for multipart string values

**Review by sradack — APPROVED** [review]

**Comment by sradack** [comment]
I spoke with @pschorf about this.  He is fine with my merging it.

**Pull request merged**